src/Microsoft.OData.Core/UriParser/SemanticAst/OperationImportSegment.cs

```diff
@@ -111,18 +111,18 @@ public OperationImportSegment(IEnumerable<IEdmOperationImport> operationImports,
 
             // Determine the return type of the operation. This is only possible if all the candidate operations agree on the return type.
             // TODO: Because we work on types and not type references, if there are nullability differences we'd ignore them...
-            IEdmType typeSoFar = this.operationImports.First().Operation.ReturnType != null
-                                     ? this.operationImports.First().Operation.ReturnType.Definition
+            IEdmType typeSoFar = this.operationImports.First().Operation.Return?.Type != null

```

Comment:
Curious about invocation of `this.operationImports.First()` twice. Does it introduce an extra cost? Should we perhaps assign `this.operationImports.First()` to a variable to mitigate that?